### DISCUSSION TITLE
Blazor Hybrid API calls developer tools network monitoring

### DISCUSSION THREAD

**Discussion by candritzky** [dotnet/maui#16191] [Q&A]
I have a WPF Blazor Hybrid app (using BlazorWebView) that calls AddBlazorWebViewDeveloperTools to allow for opening the browser dev tools via F12. This allows me to see and debug the HTML elements, CSS styles etc.

But the Network tab doesn't show any network traffic for API calls sent by my application. I assume this is because the application is running on .NET (Desktop) and uses .NET's native HttpClient and not the browser's fetch API.

Is there a way to connect these, e.g. by providing a HttpMessageHandler that routes network traffic through the browser's HTTP stack, at least for development/debugging purposes?

Or maybe the other way round: "Smuggle" the native HttpClient calls into the network tab of the browser's dev tools? The goal is to see the network traffic (without having to use an external HTTP debugging proxy), not to use the browser's HTTP stack. It should be as convenient as developing a regular SPA Web app.

**Comment by keenfooter0x315** [comment]
I need solutions, too
